pmtiles/server.go

```diff
@@ -91,6 +94,14 @@ func (server *Server) Start() {
 		totalSize := 0
 		ctx := context.Background()
 
+		cacheSize := prometheus.NewGauge(prometheus.GaugeOpts{
+			Namespace: "pmtiles",
+			Subsystem: "cache",
+			Name:      "size",
+			Help:      "Current number or directories in the cache",
+		})
+		prometheus.MustRegister(cacheSize)
```

Comment:
One concern is that these are all global and there can't be name conflicts, so this will fail if multiple servers are created. That can't happen with the CLI but can when used as a library and maybe when the caddy plugin is used? If we expect multiple servers to be a common pattern then is there a unique identifier we can use to deduplicate them? The fallback would be to require callers to set a unique metrics key.

Also monitoring these values will add a small performance overhead, if we want to eliminate that when metrics aren't being used then we could wrap these into a utility that does nothing when metrics aren't needed.

Reply:
Also MustRegister panics if it fails (ie duplicate metric name). We could make it just log an error in that case ?

Reply:
I wouldn't worry about multiple servers in the same process for now

Reply:
👍 changed it to just log a warning if the metric fails to register, so anyone who is using multiple server instances won't get a fatal error